Modules/_asynciomodule.c

```diff
@@ -1576,7 +1679,9 @@ FutureIter_am_send(futureiterobject *it,
     }
 
     it->future = NULL;
+    Py_BEGIN_CRITICAL_SECTION(fut);
```

Comment:
The accesses to `fut->fut_state` and `fut->fut_blocking` are now outside the lock.

What's the purpose of setting `it->future` to `NULL` here?

Reply:
> What's the purpose of setting it->future to NULL here?

I think it is redundant, this code is very old so maybe it was needed at some point but I don't think it is needed now.

Reply:
done